WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPhotoViewerActivity.java

```diff
@@ -279,6 +281,7 @@ public void restoreState(Parcelable state, ClassLoader loader) {
             }
         }
 
+        @NonNull
```

Comment:
Since `newInstance` is analyzed and confirmed to be `@NotNull`, I wonder if the annotations should be applied there as well. Wdyt?

Reply:
Ah yea, I can do that, nice catch, but then again, I wonder how deep I should go with that? 🤔 

FYI: I have arbitrarily (usually) stopped at the first level, when working on this PR, with the plan being that this annotations at this level will then help with other follow-up PRs where will be focusing on a case-by-case scenario, like adding missing annotations to `ReaderPhotoViewerFragment.java` or any other strategy we chose to follow going forward.

Reply:
>  I wonder how deep I should go with that? 

I had assumed (perhaps wrongly so 😅), that you already _did_ analyze the return value of this static factory method. It reads like a constructor, but unfortunately, unlike a constructor, does not have any guarantees about the return type being `@NonNull`. But, a quick look at the method suggests so, and I thought that is how you came to annotate this as such.

But reflecting on this now, I see that it is coming from the other way around, that you annotated this return type as `@NotNull`, to match the super type signature. I think this is fine, since we already _know_ it was working (and we also know it can't be null). When I reviewed this, I checked the return type of the `newInstance` call to verify it was appropriately `@NonNull`, so in this case it would also be good to annotate that as well, but feel free to defer to another PR for that.

Reply:
Yea @mkevins , these kind of PRs and work, unless being discussed and agreed ahead of time, it is hard to make assumptions on how one would approach such work, or how was something that is already completed have been built to completion and why. 🤷 

> When I reviewed this, I checked the return type of the newInstance call to verify it was appropriately @NonNull, so in this case it would also be good to annotate that as well, but feel free to defer to another PR for that.

As explained above, I will indeed prefer to defer that to another PR just to have this or anything other such super type signature method matching PR progress faster, both author and review wise. 👍